appserver/extras/payara-micro/payara-micro-core/src/main/java/fish/payara/micro/impl/PayaraMicroImpl.java

```diff
@@ -1077,6 +1056,13 @@ public PayaraMicroRuntime bootStrap() throws BootstrapException {
         }
     }
 
+    private void callHandler(Consumer<AdminCommandRunner> handler) throws GlassFishException {
+        CommandRunner runner = gf.getCommandRunner();
+        if (handler != null) {
+            handler.accept((cmd, args) -> new CommandResultWrapper(runner.run(cmd, args)));
```

Comment:
Don't know how the dependencies look like but it might be possible to have `CommandRunner` implement `AdminCommandRunner` and have `CommandResult` implement `ClusterCommandResult`.

Reply:
Unfortunately those are part of Embedded API, and we don't want embedded to be dependant on Micro, or vice versa. Also that embedded API is public API since Glassfish 3 and adding inheritance would break compatibility. Especially problematic one is `CommandResult.ExitStatus` which is an enum.